geonetwork/gn_api.py

```diff
@@ -184,5 +184,26 @@ def delete_thesaurus_dict(self, name):
         response.raise_for_status()
         return response.json()
 
+    def search(self, query):
+        """
+        Use geonetwork API to search metadata
+        :param query: query generated by frontend app like datahub of geonetwork
+        looks like that :
+         {"query":{"bool":{"must":[{"terms":{"isTemplate":["n"]}},{"multi_match":{"query":"test","type":"bool_prefix","fields":["resourceTitleObject.*^4","resourceAbstractObject.*^3","tag^2","resourceIdentifier"]}}],"must_not":{"terms":{"resourceType":["service","map","map/static","mapDigital"]}}}},"_source":["resourceTitleObject","uuid"],"from":0,"size":20}
+        """
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type' : 'application/json'
+        }
+        url = self.api_url + "/search/records/_search?bucket=bucket"
+        print(query)
```

Comment:
please remove print statetment, there is a complete logging mechanism where each session request is logged if log level is DEBUG (by default these logs goes to stdout)
https://github.com/camptocamp/python-geonetwork/blob/82e3b49a80512e64e826947da3f2e79ab5cb756b/geonetwork/gn_session.py#L57